src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +546,118 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
-        private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
+        private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellationToken)
         {
-            Dictionary<string, ConfigurationSetting> data = null;
-            Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-            
+            var startupExceptions = new List<Exception>();
+
+            int attempts = 0;
+
             try
             {
-                await ExecuteWithFailOverPolicyAsync(
-                    availableClients,
-                    async (client) =>
+                while (true)
+                {
+                    attempts++;
+
+                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetClients();
```

Comment:
We can call `GetClients` outside the while loop

Reply:
Do we expect the clients to change between retries, for example, with replica auto-discovery?

Reply:
Oh that's a good point, it could change.